private/runtime/binding-spec.rkt

```diff
@@ -224,7 +231,9 @@
        (define scoped-transformer-stx (add-scopes transformer-stx local-scopes))
        (let ([bound-id (bind-and-record-rename! (add-scopes id local-scopes) #`(#,constr-id #,(flip-intro-scope scoped-transformer-stx)) space)])
          (values scoped-transformer-stx bound-id)))]
-    [(bind-syntaxes pv space constr-id transformer-pv)
+    [(bind-syntaxes depth pv space constr-id transformer-pv)
+     (unless (= 1 depth)
+       (error "don't know how to handle depth > 1 yet"))
```

Comment:
I don't think we need an error like this, in either the explicit or implicit case. With explicit, depth within the bind-syntaxes should be = 1, and externally there might be ellipses.

So I think we don't even need the `depth` field in the runtime. In a use of `bind-syntaxes`, the pvar should always be referring to a pvar binding with a binding class associated, not a nonterminal.

With implicit, it seems like the correct behavior should be to assume that bind-syntaxes consumes one depth internally, and repeats overall for additional depth. Based on example (my-letrec-syntaxes), it seems like that's what would happen previously. And the way the for/pv-state-tree works below should accomplish that.